_posts/2015-06-29-implementing-a-css-style-guide.markdown

```diff
@@ -0,0 +1,87 @@
+    - A good exception to this rule would be for the use of pseudo selectors in the vein of `nth-child` and `nth-of-type`.
+
+### Variables
+- Make use of our `z-index` scale file. 
+    - The scale file has 10 layers of private variables.
+    - Every time a z-index is used, create a new variable that identifies its value and the feature being used (i.e. `@zIndex-4-profileAvater`).
+    - This file helps preserve sanity when it comes to z-indexes, and allows us to look in a single file to understand different layers of the application.
+- Use our own line-height, font-weight, and letter-spacing scale files.
+- Color variables and font variables should be declared in separate files and referenced whenever possible.
+
+### General Style
+- Comma seperated selectors should each be declared on a new line.
+- Include exactly one line break after each selector block.
+- Use single quotes for urls.
+- Use the font shorthand when possible (`font: 14px $proximanova $black`).
+- `@include`s (for breakpoints) should be declared at the top of a given breakpoint-specific stylesheet, and should wrap all breakpoint-specific styles.
+- Always use `rgb` and `rgba` for colors, to avoid mixing hex codes and `rgba` for instances where transparency is needed.
+
+### Grid
+- Elements should always be on the established grid (the assumption being that the design team will always place elements in the context of a grid going forward.)
+- Horizontal padding and margins should be of some factor of our established gutter widths.
+
+### Breakpoints
+- Place breakpoint declarations inside a new file and folder `BREAKPOINT_NAME/component_name`, per the file structure above (e.g. `tablet/_load_more_button.scss`).
+- As much as possible, avoid duplication of shared styles across different breakpoints.
+
+---
+
+## Lessons Learned
+
+The hard part, maybe not surprisingly, has been actually implementing the rules we prescribed above. So in addition to sharing our style guide, I also wanted to share a few things I learned in implementing it, which will hopefully help anyone out there who is thinking about implementing a CSS style guide of their own.
+
+### Be vigilant
+
+I've learned the hard way the importance of getting CSS code into as good a state as possible before calling a feature "done". We've all said to ourselves "I'm just going to get this to look right, and then I'll come back and refactor later." Except that you won't. Because refactoring CSS is terrible. Class names tend to be quite "sticky" and resistent to refactoring given the way they get reused, and after giving in to a sub-optimal solution in order to fix a bug quickly, changing things can feel precarious. This is why it's worth spending a little extra time up front to make sure you are following your own rules (harder to do than I thought). The same goes for reviewing code from other team members - it's worth being a little nit picky on the initial pull request for the sake of everyone's future sanity.
```

Comment:
I would add some emphasis around the lines
_Except that you won't. Because *refactoring CSS is terrible*._